### ISSUE TITLE
Autohide panels only when a window is maximized

### ISSUE THREAD

**Issue by kennyj2369** [mate-desktop/mate-panel#340]
I have two MATE panels configured - one on top and one on the bottom. Right now I have them set to autohide but I really only want this to happen if I have a window maximized. If I minimize or restore a window, I'd like the panels to be visible. I asked about this on another website and was told it would have to be a feature request.

**Comment by NiceandGently** [comment]
Why should a panel not hide if non window is maximize?
If a user like this feature, maybe on small notebook displays, we will introduce a bug.

**Comment by gurpal2000** [comment]
@kennyj2369 isn't this the same as saying "allow windows to cover the panels" ? Or is there something subtle here?

[Background: I have this problem where i run virtualbox in full screen mode across dual screens in Linux Mint. Even with MATE's autohide or Cinnamon's smart hide, the panels do start appearing sometimes when clicking on different screens. KDE/Plasma 5 manages this fine, so i'm assuming its a MATE/panel issue.]

**Comment by kennyj2369** [comment]
Yes @gurpal2000 I suppose you could phrase it that way. Although, I would like to be able to panel by moving my mouse to the top of the screen similar to how autohide currently works. 

@raveit65 If the window is not maximized then I am likely moving between applications and possibly doing more than one thing at a time. I think in that instance, I would prefer the panels be visible to allow me to keep track of things better.  I don't think this should be considered a bug (as in an error in the coding).  I think this should be considered a feature request.

**Comment by rumly111** [comment]
In Unity there was a similar behavior (as described), I'd like to see it in Mate as well.
